index.js

```diff
@@ -111,7 +111,23 @@ class Tonic extends window.HTMLElement {
     }
   }
 
-  _set (target, content = '') {
+  async _set (target, render, text = '') {
+    const name = render && render.constructor.name
+    let content = ''
+
+    if (name === 'AsyncFunction') {
```

Comment:
instead of checking name we could more reliably check instance, 

You can get constructors with:

```js
const { constructor: AsyncFunction } = async function () {}
const { constructor: AsyncFunctionGenerator } = async * function () {}
```

then:

```js
if (render instance of AsyncFunction) {
```
 
and so on

Reply:
ah yeah! :+1: